internal/interpreters/python_requires.go

```diff
@@ -0,0 +1,90 @@
+package interpreters
+
+// Copyright (C) 2025 by Posit Software, PBC.
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/posit-dev/publisher/internal/util"
+
+	"gopkg.in/ini.v1"
+
+	toml "github.com/pelletier/go-toml/v2"
+)
+
+type PyProjectPythonRequires struct {
+	ProjectPath util.AbsolutePath
+}
+
+func NewPyProjectPythonRequires(projectPath util.AbsolutePath) *PyProjectPythonRequires {
+	return &PyProjectPythonRequires{
+		ProjectPath: projectPath,
+	}
+}
+
+func (p *PyProjectPythonRequires) GetPythonVersion() (string, error) {
+	if version, err := p.readPythonVersionFile(); err == nil && version != "" {
+		return version, nil
+	}
+	if version, err := p.readPyProjectToml(); err == nil && version != "" {
+		return version, nil
+	}
+	if version, err := p.readSetupCfg(); err == nil && version != "" {
+		return version, nil
+	}
+	return "", errors.New("no python version requirement found")
+}
+
+func (p *PyProjectPythonRequires) readPythonVersionFile() (string, error) {
```

Comment:
They look pretty simple, but would you mind adding some comments describing the format expected in each of these files (maybe an example)?